### PR TITLE
add spectrum mask to desi

### PR DESCRIPTION
**Pull request by al-jshen**
Adds a spectrum mask to DESI indicating whether a pixel in a spectrum is bad. Follows the SDSS convention of mask True=bad pixel. This mask is just a boolean which is turned on if any bit in the DESI SPECMASK is on, or if the ivar is below 1e-6 (following SDSS).

### CODE REVIEW

**Comment by al-jshen** [comment]
tests fail because of this https://github.com/huggingface/datasets/pull/6976 (numpy 2.0 breaks some things in HF datasets, they fixed it a couple hours ago but the pip install in the GH action runner still uses the old version of HF datasets)

File: scripts/desi/build_parent_sample.py [hunk 1]
```diff
@@ -82,76 +100,109 @@ def save_in_standard_format(args):
...
+            args.output_dir,
+            "edr_sv3/healpix={}/001-of-001.hdf5".format(group["healpix"][0]),
+        )
         map_args.append((group, group_filename, args.desi_data_path))
 
     # Run the parallel processing
-    with Pool(args.num_processes) as pool:
-        results = list(tqdm(pool.imap(save_in_standard_format, map_args), total=len(map_args)))
+    # with Pool(args.num_processes) as pool:
+    #     results = list(tqdm(pool.imap(save_in_standard_format, map_args), total=len(map_args)))
```
Comment (EiffL) [inline scripts/desi/build_parent_sample.py:178]:
```suggestion
```

File: scripts/desi/build_parent_sample.py [hunk 2]
```diff
@@ -82,76 +100,107 @@ def save_in_standard_format(args):
...
+        group_filename = os.path.join(
+            args.output_dir,
+            "edr_sv3/healpix={}/001-of-001.hdf5".format(group["healpix"][0]),
+        )
         map_args.append((group, group_filename, args.desi_data_path))
 
     # Run the parallel processing
-    with Pool(args.num_processes) as pool:
-        results = list(tqdm(pool.imap(save_in_standard_format, map_args), total=len(map_args)))
+
```
Comment (EiffL) [inline scripts/desi/build_parent_sample.py:177]:
```suggestion
```

**Review by EiffL — APPROVED** [review]
It looks good to me

**Pull request merged**